Issue 240 add shelter admin

Taktar opened the pull request:
Fixes #240

**What was changed?**

- A new `AddUserForm.jsx` React component was added under `components/shelter/`.
- A corresponding CSS file `AddUserForm.css` was created under `styles/shelter/`.
- `ShelterDashboard.js` was updated to display the `AddUserForm` via a settings button on the page.

**Why was it changed?**

The change supports the system's authorization policy, which allows a shelter admin to promote another user to shelter admin status. This UI functionality allows shelter admins to add new admins directly from the dashboard, fulfilling the requirement in Issue #240.

**How was it changed?**

- `AddUserForm.jsx` captures user email input and sends a POST request to `/user_permission` with the correct `resource_type`, `resource_id`, and `user_email`.
- `ShelterDashboard.js` was updated minimally to include a "Settings" button and conditionally render the `AddUserForm` component.
- All code follows the existing structure and design pattern of the app.

Taktar commented:
<img width="1286" alt="Screenshot 2025-04-07 at 2 59 22 PM" src="https://github.com/user-attachments/assets/0d352717-5bb8-4ebd-b4e8-54b296ab35ba" />

strejoarciles commented on client_app/src/components/shelter/ShelterDashboard.js:
```diff
@@ -114,12 +116,36 @@ function ShelterDashboard({ shelterId }) {
+                  padding: "0",
+                  marginTop: "-5px"
+                }}
+                onClick={() => setShowAddUserForm(!showAddUserForm)}
```
The onClick toggle for showing/hiding the <AddUserForm /> is concise and readable.

strejoarciles commented on client_app/src/components/shelter/ShelterDashboard.js:
```diff
@@ -11,11 +11,13 @@ import setMilliseconds from "date-fns/setMilliseconds";
   const [shiftDetails, setShiftDetails] = useState([]);
   const [showAllPastVolunteers, setShowAllPastVolunteers] = useState(false);
   const [showAllTodaysShifts, setShowAllTodaysShifts] = useState(false);
+  const [showAddUserForm, setShowAddUserForm] = useState(false); 
```
Since the button is toggling a form’s visibility, adding aria-expanded={showAddUserForm} would improve screen reader support.

kate-holdener commented:
> I am struggling with this: <img alt="Screenshot 2025-04-07 at 2 57 36 PM" width="1427" src="https://private-user-images.githubusercontent.com/157086064/431097814-6fc57b26-2573-4f19-b81c-a0ed1c3cecfc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQyOTQ5ODMsIm5iZiI6MTc0NDI5NDY4MywicGF0aCI6Ii8xNTcwODYwNjQvNDMxMDk3ODE0LTZmYzU3YjI2LTI1NzMtNGYxOS1iODFjLWEwZWQxYzNjZWNmYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMFQxNDE4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYmI0YmMzMjRkMjgzOWJmOTc1NjJmYjA1MDk5MDlmYWQyZTM3MTlmNTE2ODY5MGFmYzVmNzU0N2VmNmI5OWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6zTEVnr4tIv-EJKhcoxqY06dBZRawImgYoSfOWAWtJg">
> 
> <img alt="Screenshot 2025-04-07 at 2 57 59 PM" width="1416" src="https://private-user-images.githubusercontent.com/157086064/431097942-811ce2e3-89da-4bce-a075-5cbf9a5151ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQyOTQ5ODMsIm5iZiI6MTc0NDI5NDY4MywicGF0aCI6Ii8xNTcwODYwNjQvNDMxMDk3OTQyLTgxMWNlMmUzLTg5ZGEtNGJjZS1hMDc1LTVjYmY5YTUxNTFlZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMFQxNDE4MDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYjJkNmQ5YjMxYTc3YjcyNTkwODBmYzU0YWNjMmEzNjgyMmMxYzhiYTQ2MTJmNGVlYTY5ZGMxMWMwZjg3YTgxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.bNrQFoC4AUHwD4jboZsvUJExNnfp8aw7RalDq0tPq1A">

Use our admin_cli to give your user email whatever permissions it needs: https://github.com/oss-slu/shelter_volunteers/pull/229

kate-holdener reviewed (changes requested):
@Taktar When I use your new form to add an admin for a shelter, I get an error: "Resource ID is required for shelter". The Resource ID is the shelter ID, for which we are trying to add an admin. Digging through the code, I added a console.log to print the shelterId and it shows up as undefined. It also shows up as undefined in the ShelterDashboard component. The ShelterDashboard component gets called from App.js as:
```
        <Route path="/shelter-dashboard/:shelterId" element={<ShelterDashboard />} />
```
I believe this call needs to be changed, to pass shelterId:
```
        <Route path="/shelter-dashboard/:shelterId" element={<ShelterDashboard shelterId={shelterId}/>} />
```
I added some more inline comments for improving your solution.

It's a good start, by adding the link to add new admins to the dashboard. Once you get it working, please move this functionality into its own "Settings" screen, as described in issue #240

Taktar commented:
<img width="897" alt="Screenshot 2025-04-10 at 4 36 14 PM" src="https://github.com/user-attachments/assets/3e7860c3-777b-4ca1-b4d3-1d82806a92c2" />
I am getting login errors when I open up http://localhost:3000/

Does this mean I am not redirected properly?

Taktar commented:
Added /shelter-dashboard/:shelterId/settings route in App.js

Created Settings.js component to display the AddUserForm

Linked to the Settings page from the ShelterDashboard

Reused AddUserForm for adding/removing shelter admins

Used CLI to assign shelter admin permissions for testing

Verified functionality by signing up and logging in with the test admin account

kate-holdener commented:
> <img alt="Screenshot 2025-04-10 at 4 36 14 PM" width="897" src="https://private-user-images.githubusercontent.com/157086064/432509099-3e7860c3-777b-4ca1-b4d3-1d82806a92c2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMjk2NTAsIm5iZiI6MTc0NDMyOTM1MCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTA5MDk5LTNlNzg2MGMzLTc3N2ItNGNhMS1iNGQzLTFkODI4MDZhOTJjMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMFQyMzU1NTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZTg2NDQ5ZDljODJmNWUwZTczODY0NjhkM2NjMTdhZTQ3ZTM3YzJjY2Y1ZTEzZThjN2FiZmI2OGMyZDg4M2IxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.zE80VDeMG1wboFeTrDc2CXztRlMiXLLycou80tMhIis"> I am getting login errors when I open up http://localhost:3000/
> Does this mean I am not redirected properly?

What credentials are you using to log in? Can you post a screenshot of the errors?

Taktar commented:
<img width="1430" alt="Screenshot 2025-04-10 at 8 27 00 PM" src="https://github.com/user-attachments/assets/10f01300-b402-4454-af9f-5869029656f0" />
I did this and I also tried shelteradmin@example.com with 12345 as the password

kate-holdener commented:
> <img alt="Screenshot 2025-04-10 at 8 41 05 PM" width="557" src="https://private-user-images.githubusercontent.com/157086064/432568229-74264ce6-61a0-4c41-94c6-b527a417b3c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MjI5LTc0MjY0Y2U2LTYxYTAtNGM0MS05NGM2LWI1MjdhNDE3YjNjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2RiOGMyNWZkZGQwM2ZmYTY4MzExNjMzNGUwZWU4NDNiMzZiZWNjMTgyMDYxN2FlM2NiZTMyNDgzNjI4MjAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HRfjALq1ZjfCa-JCCLKx7NxxaAsXqbJPiIJR9B9bbG4"> <img alt="Screenshot 2025-04-10 at 8 41 46 PM" width="1355" src="https://private-user-images.githubusercontent.com/157086064/432568351-d9b49d48-d673-4a51-958b-88de53aab863.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MzUxLWQ5YjQ5ZDQ4LWQ2NzMtNGE1MS05NThiLTg4ZGU1M2FhYjg2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzZhNGNiOWM3YWQ1NzE3MDczYTI3MGQwNTNkZmU5ZDY5MWE5MTZiNTQyMjNhY2E4YzhjMjNiMmRiMGZjNDhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4PjxK8mqAE5bPfmnY5V9Zg5rf8hF7u6tuYfuEtkLyuE">

We handle authentication (validating the user's credentials) and authorization (checking what resources the user has access to) separately. When you gave this user shelter admin role, it didn't add the user to the system. The admin_cli simply said: this user (if this user ever authenticates) will be able to be an admin for this shelter. This user, however, does not exist in our fake authentication system (we'll replace our fake system with google's authentication later). All existing users are in `server/authentication/user_list.json`. You can take either one of those user emails and give that user shelter admin access, or add your user email to that list.

Taktar commented:
> > <img alt="Screenshot 2025-04-10 at 8 41 05 PM" width="557" src="https://private-user-images.githubusercontent.com/157086064/432568229-74264ce6-61a0-4c41-94c6-b527a417b3c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MjI5LTc0MjY0Y2U2LTYxYTAtNGM0MS05NGM2LWI1MjdhNDE3YjNjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2RiOGMyNWZkZGQwM2ZmYTY4MzExNjMzNGUwZWU4NDNiMzZiZWNjMTgyMDYxN2FlM2NiZTMyNDgzNjI4MjAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HRfjALq1ZjfCa-JCCLKx7NxxaAsXqbJPiIJR9B9bbG4"> <img alt="Screenshot 2025-04-10 at 8 41 46 PM" width="1355" src="https://private-user-images.githubusercontent.com/157086064/432568351-d9b49d48-d673-4a51-958b-88de53aab863.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MzUxLWQ5YjQ5ZDQ4LWQ2NzMtNGE1MS05NThiLTg4ZGU1M2FhYjg2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzZhNGNiOWM3YWQ1NzE3MDczYTI3MGQwNTNkZmU5ZDY5MWE5MTZiNTQyMjNhY2E4YzhjMjNiMmRiMGZjNDhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4PjxK8mqAE5bPfmnY5V9Zg5rf8hF7u6tuYfuEtkLyuE">
> 
> We handle authentication (validating the user's credentials) and authorization (checking what resources the user has access to) separately. When you gave this user shelter admin role, it didn't add the user to the system. The admin_cli simply said: this user (if this user ever authenticates) will be able to be an admin for this shelter. This user, however, does not exist in our fake authentication system (we'll replace our fake system with google's authentication later). All existing users are in `server/authentication/user_list.json`. You can take either one of those user emails and give that user shelter admin access, or add your user email to that list.
> 
> Creating an account functionality currently does not work. We will be removing it (since users will be signing in with their Google accounts)

I did try developer@slu.edu already

kate-holdener commented:
> > <img alt="Screenshot 2025-04-10 at 8 41 05 PM" width="557" src="https://private-user-images.githubusercontent.com/157086064/432568229-74264ce6-61a0-4c41-94c6-b527a417b3c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MjI5LTc0MjY0Y2U2LTYxYTAtNGM0MS05NGM2LWI1MjdhNDE3YjNjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2RiOGMyNWZkZGQwM2ZmYTY4MzExNjMzNGUwZWU4NDNiMzZiZWNjMTgyMDYxN2FlM2NiZTMyNDgzNjI4MjAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HRfjALq1ZjfCa-JCCLKx7NxxaAsXqbJPiIJR9B9bbG4"> <img alt="Screenshot 2025-04-10 at 8 41 46 PM" width="1355" src="https://private-user-images.githubusercontent.com/157086064/432568351-d9b49d48-d673-4a51-958b-88de53aab863.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MzUxLWQ5YjQ5ZDQ4LWQ2NzMtNGE1MS05NThiLTg4ZGU1M2FhYjg2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzZhNGNiOWM3YWQ1NzE3MDczYTI3MGQwNTNkZmU5ZDY5MWE5MTZiNTQyMjNhY2E4YzhjMjNiMmRiMGZjNDhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4PjxK8mqAE5bPfmnY5V9Zg5rf8hF7u6tuYfuEtkLyuE">
> 
> We handle authentication (validating the user's credentials) and authorization (checking what resources the user has access to) separately. When you gave this user shelter admin role, it didn't add the user to the system. The admin_cli simply said: this user (if this user ever authenticates) will be able to be an admin for this shelter. This user, however, does not exist in our fake authentication system (we'll replace our fake system with google's authentication later). All existing users are in `server/authentication/user_list.json`. You can take either one of those user emails and give that user shelter admin access, or add your user email to that list.
> 
> Creating an account functionality currently does not work. We will be removing it (since users will be signing in with their Google accounts)

Make s

> > > <img alt="Screenshot 2025-04-10 at 8 41 05 PM" width="557" src="https://private-user-images.githubusercontent.com/157086064/432568229-74264ce6-61a0-4c41-94c6-b527a417b3c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MjI5LTc0MjY0Y2U2LTYxYTAtNGM0MS05NGM2LWI1MjdhNDE3YjNjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04M2RiOGMyNWZkZGQwM2ZmYTY4MzExNjMzNGUwZWU4NDNiMzZiZWNjMTgyMDYxN2FlM2NiZTMyNDgzNjI4MjAyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HRfjALq1ZjfCa-JCCLKx7NxxaAsXqbJPiIJR9B9bbG4"> <img alt="Screenshot 2025-04-10 at 8 41 46 PM" width="1355" src="https://private-user-images.githubusercontent.com/157086064/432568351-d9b49d48-d673-4a51-958b-88de53aab863.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzMzk2NjgsIm5iZiI6MTc0NDMzOTM2OCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyNTY4MzUxLWQ5YjQ5ZDQ4LWQ2NzMtNGE1MS05NThiLTg4ZGU1M2FhYjg2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQwMjQyNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYzZhNGNiOWM3YWQ1NzE3MDczYTI3MGQwNTNkZmU5ZDY5MWE5MTZiNTQyMjNhY2E4YzhjMjNiMmRiMGZjNDhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4PjxK8mqAE5bPfmnY5V9Zg5rf8hF7u6tuYfuEtkLyuE">
> > 
> > 
> > We handle authentication (validating the user's credentials) and authorization (checking what resources the user has access to) separately. When you gave this user shelter admin role, it didn't add the user to the system. The admin_cli simply said: this user (if this user ever authenticates) will be able to be an admin for this shelter. This user, however, does not exist in our fake authentication system (we'll replace our fake system with google's authentication later). All existing users are in `server/authentication/user_list.json`. You can take either one of those user emails and give that user shelter admin access, or add your user email to that list.
> > Creating an account functionality currently does not work. We will be removing it (since users will be signing in with their Google accounts)
> 
> I did try [developer@slu.edu](mailto:developer@slu.edu) already

Make sure your server side application is running. The login will be sending requests to the server. If it is running, do you see any messages in that terminal (the one running the server).

Taktar commented:
thanks so much! Now I am able to login but this is what my screen looks like. Did I mess things up?
<img width="1420" alt="Screenshot 2025-04-11 at 10 06 25 AM" src="https://github.com/user-attachments/assets/168b262d-4a1e-4339-9296-71abbdd67263" />
<img width="1437" alt="Screenshot 2025-04-11 at 10 06 39 AM" src="https://github.com/user-attachments/assets/28873c07-c098-4a8e-8cee-bac5d047403e" />

kate-holdener commented:
> thanks so much! Now I am able to login but this is what my screen looks like. Did I mess things up?
> <img alt="Screenshot 2025-04-11 at 10 06 25 AM" width="1420" src="https://private-user-images.githubusercontent.com/157086064/432818782-168b262d-4a1e-4339-9296-71abbdd67263.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzODkzNjAsIm5iZiI6MTc0NDM4OTA2MCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyODE4NzgyLTE2OGIyNjJkLTRhMWUtNDMzOS05Mjk2LTcxYWJiZGQ2NzI2My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQxNjMxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mMjU0MTM0ZmU5NWMwMDVmOGJlYzc3ZDdlNDYxODYxMDRkNjA0MjYzZGZlNDc2YWY0ZmRmNWFmMjg0MGZmODAwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.jUblYqycRzWz9uauZKkj2SEgjk4LtZtPifjwb9ulONs">
> <img alt="Screenshot 2025-04-11 at 10 06 39 AM" width="1437" src="https://private-user-images.githubusercontent.com/157086064/432818854-28873c07-c098-4a8e-8cee-bac5d047403e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQzODkzNjAsIm5iZiI6MTc0NDM4OTA2MCwicGF0aCI6Ii8xNTcwODYwNjQvNDMyODE4ODU0LTI4ODczYzA3LWMwOTgtNGE4ZS04Y2VlLWJhYzVkMDQ3NDAzZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNDExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDQxMVQxNjMxMDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOWIwNDE5ZTRhOWZjMWIyNGI5ZWNlMzYyNzlhYTQ5NGE5N2RiYjcwODM5ZDQ0ZDM2NDNlNTFmOTg1YjFiMWYxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GFu1y0CJIT0wh9gFbUGHcHj8Q_7G8G42_pdi4q8mNDo">

It looks reasonable. If you give your user that you logged in as (developer@slu.edu?) system admin rights, you'll see a different set of dashboard options when you log in. Basically, the user sees whatever they have access to. The "S" in your volunteer dashboard is the shelter you have created in your database.

Taktar commented:
I am confused on how to give admin rights or certain access to stuff. Is that going to be done on POSTman?

kate-holdener commented:
> I am confused on how to give admin rights or certain access to stuff. Is that going to be done on POSTman?

Use admin_cli.py to make developer@slu.edu a system admin. This will give developer@slu.edu access to everything.

kate-holdener commented on client_app/src/components/shelter/ShelterDashboard.js:
```diff
@@ -114,12 +117,43 @@ function ShelterDashboard({ shelterId }) {
             </div>
             <PastVolunteersContainer shiftDetails={shiftDetails} />
           </div>
+          {/* Existing inline AddUserForm section */}
+          <div className="container-small">
+            <div className="container-align">
+              <h4>Settings</h4>
+              <button
+                style={{
+                  backgroundColor: "#f9f6f6",
+                  border: "none",
+                  outline: "none",
+                  color: "#1F75FE",
+                  fontSize: "1.0rem",
+                  textDecoration: "underline",
+                  textAlign: "center",
+                  padding: "0",
+                  marginTop: "-5px"
+                }}
+                onClick={() => setShowAddUserForm(!showAddUserForm)}
```
I don't think we'll want the inline AddUserForm in the Dashboard. This will not be a frequently used option.

kate-holdener reviewed (changes requested):
Adding the admin functionality seems to work. When the user is added successfully, the UI presents an info message that the user was added. The background color of that message is red/pink, making the user think that there was an error. Please change that color to green.
<img width="1704" alt="Screenshot 2025-04-11 at 12 49 40 PM" src="https://github.com/user-attachments/assets/526981cc-519c-4b21-be91-67f01668a1d5" />

Also, it would be good to have "Add Shelter Admin" as an expandable option in the Settings (kind of like what you have on the Shelter Dashboard right now). But that's a "nice to have", not a "must have"

Taktar commented:
Removed the inline AddUserForm section from ShelterDashboard.js as requested

Added a link to the new /shelter-dashboard/:shelterId/settings route for managing users

Cleaned up unused state (showAddUserForm) and removed the AddUserForm import

Updated .message.success style in AddUserForm.css to use a green background for clearer success feedback

Taktar commented:
> It still shows up red for me. <img alt="Screenshot 2025-04-14 at 12 58 39 PM" width="846" src="https://private-user-images.githubusercontent.com/42073871/433494590-3af72666-a8ec-4481-a6d3-4256a3d12033.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ2NTM5OTAsIm5iZiI6MTc0NDY1MzY5MCwicGF0aCI6Ii80MjA3Mzg3MS80MzM0OTQ1OTAtM2FmNzI2NjYtYThlYy00NDgxLWE2ZDMtNDI1NmEzZDEyMDMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE0VDE4MDEzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZiNDlkNTljODAwMTA3YjA4M2Y0MTE4MDkzODE4ODZkMjY2NjE2MTkwZmY5ODFmMzljMTJmMTFjNjExMTU5Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3gn2Fo4zw4DTfKVEuQAyUwN2cac9btDwZ0_BwgxzVe4">

I think I edited the wrong css file. I was trying to look trhough all of this. Do you know which file has this part with the pink?
